Add Download Button for Code Snippets

iwasawamnb opened the pull request:
*Issue #, if available:*

*Description of changes:*
Added functionality to display a download button when an LLM's response contains a code snippet. This allows users to conveniently download the code as a file for easier use.
![image](https://github.com/user-attachments/assets/092664fd-5af7-4c59-9a76-272e0a47494c)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Commit:
added ButtonDownload to code snippet

statefb reviewed (commented):
Could you add `ButtonDownload.story.tsx` and check if it's displayed properly on Ladle? (run `npm run ladle` on frontend directory)

Commit:
added ButtonDownload.stories.tsx

iwasawamnb commented:
@statefb 
Thank you for your feedback. I have added `ButtonDownload.story.tsx` and confirmed that it is displayed correctly when running `npm run ladle`.

Additionally, I encountered the following error when running `npm run ladle` with the original code. To address this, I renamed `frontend/tailwind.config.cjs`. Could you please verify if this change is acceptable?

Error:
```
  plugins: [require('@tailwindcss/typography'), require('tailwind-scrollbar')],
           ^

ReferenceError: require is not defined
    at file:///.../bedrock-claude-chat/frontend/tailwind.config.js:61:12
```

statefb commented:
@iwasawamnb Thank you for adding the story. Regarding `config.cjs`, I could not reproduce your problem on Node `v20.10.0`. Could you re-run `npm ci` and try it again? If the config commit reverted, we'll merge this PR.

Commit:
reverted tailwind.config.js filename

iwasawamnb commented:
@statefb 
Apologies, I re-ran `npm run ladle` in the Node v20.10.0 environment, and it worked fine. Therefore, I reverted the config file name to its original.

statefb commented:
@iwasawamnb Thanks for your confirmation! Could you check lint error?
https://github.com/aws-samples/bedrock-claude-chat/actions/runs/13002697156/job/36268026615?pr=693

Commit:
fix lint error

statefb commented on frontend/src/components/ButtonDownload.tsx:
```diff
@@ -0,0 +1,88 @@
+import { BaseProps } from '../@types/common';
+import { PiDownload } from 'react-icons/pi';
+
+declare global {
```
Is it really necessary to use the global scope? If it is not required, please declare it locally. If there is a valid reason to use the global scope, include a comment in the source code explaining why.

iwasawamnb replied:
Thank you for your comment. Since the type definitions doesn't need to be global, I've modified it to be locally.

Commit:
change type definition from global to local

statefb reviewed (approved):
LGTM! Thank you for contribution!
